recommender/engine/pre_agent/pre_agent.py

```diff
@@ -25,8 +26,18 @@ def _services_to_ids(services):
     return [s.id for s in services]
 
 
+# Fallback in case of no of candidate_services being too small
+def _fill_candidate_services(candidate_services, k):
```

Comment:
what is `k`? there is no docstring, no meaningful name and no type. I can infer that it should be an `int` but that's no clear on the first glance